include/sys/trace.h

```diff
@@ -43,19 +43,23 @@
  */
 #undef DTRACE_PROBE1
 #define	DTRACE_PROBE1(name, t1, arg1) \
-	trace_zfs_##name((arg1))
+	if (arg1) \
+		trace_zfs_##name((arg1))
 
 #undef DTRACE_PROBE2
 #define	DTRACE_PROBE2(name, t1, arg1, t2, arg2) \
-	trace_zfs_##name((arg1), (arg2))
+	if (arg1 && arg2) \
```

Comment:
@loli10K Thanks for the patch. Just summarizing options we have here: 

1. Do not log an entry if there is no dbuf that can be evicted. This can also be achieved via conditional tracepoint. Looks like we are not in favor of it. 
2. Call different trace function for Null vs Non-Null dbufs. Again not preferable. 
3. Have dummy value assigned to dbuf when  it is NULL so that underlying infra doesn't complain/crash. With this, the output can be sth like:
dbuf_evict-1422  [006] ....   956.984961: zfs_dbuf__evict__one: dbuf NULL{ spa "NULL" objset 0 object 0 level 0 blkid 0 offset 0 size 0 state 0 holds 0 } or 
dbuf_evict-1422  [006] ....   956.984961: zfs_dbuf__evict__one: dbuf NULL{ spa "NULL" objset ? object ? level ? blkid ? offset ? size ? state ? holds ? } 
In this case we just end up logging some useless information. 
4.  The above patch as suggested by @loli10K that takes care of different format string at expense of snprintf. I do see the same approach being followed in multiple places in Linux kernel. Even though it involves an additional copy, it may still be ok. If we don't want that then option 3 looks to be the more favorable one.